src/main/java/org/assertj/core/util/introspection/IntrospectionStrategy.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2021 the original author or authors.
+ */
+package org.assertj.core.util.introspection;
+
+import java.util.Set;
+
+public interface IntrospectionStrategy {
+
+  Set<String> getMemberNames(Class<?> clazz);
+
+  Set<String> getMemberNamesAsFields(Class<?> clazz);
```

Comment:
My intuition tells me we should not this one, I mean if member refers to a getter/property then why would need the fields names, we would get the values of member by calling its getter.

Reply:
The idea was to try keep existing approach of addressing fields as properties, if it makes sense.
Just cause we can have getters that have nothing to do with fields 
```
public String getFullName(){
  return firstName + " " + lastName;
}
```
there is a very high chance that the compartibility will be broken. For example at the moment I cant imagine ignoring/comparing fields via string chains `"one.two.three"` when using getters comparsion.

Reply:
fair point ... that would be interpreted as `getOne().getTwo().getThree()` but I need to think if it is worth it to break the API and introduce `ignoringMembers` or something alike.